libgrust/libformat_parser/generic_format_parser/src/lib.rs

```diff
@@ -78,6 +78,7 @@ enum InputStringKind {
 }
 
 /// The type of format string that we are parsing.
+#[repr(C)]
```

Comment:
I'd like to not change this, because ideally this should be a straight up copy of rustc's `rustc_format_parser` crate. I think the best thing would be to add an extra enum on our side of the Rust crate, e.g. in `src/lib.rs` directly, which is `#[repr(C)]` and whose initial value is `= 0`. then, add a quick conversion function to `generic_format_parser::ParseMode`

Reply:
hmm i see, thanks for clearing up my confusion haha, i was just changing stuff up, good to know src/lib.rs is our side

Reply:
yeah no worries it's not super explicit!